crm_rma_prodlot_invoice/models/sale_order_line.py

```diff
@@ -28,33 +28,51 @@ class SaleOrderLine(models.Model):
 
     @api.multi
     def invoice_line_create(self):
-        res = super(SaleOrderLine, self).invoice_line_create()
-        for order_line in self:
-            prodlot_obj = self.env['stock.production.lot']
-            procurements = order_line.procurement_ids
-
-            if not procurements:
-                continue
+        invoice_lines = super(SaleOrderLine, self).invoice_line_create()
+        self._search_serial_number_to_change_from_sale(invoice_lines)
+        return invoice_lines
 
-            move_id = self.env['stock.move'].\
-                search([('procurement_id', '=',
-                         order_line.procurement_ids[0].id)], limit=1)
+    @api.multi
+    def _search_serial_number_to_change_from_sale(self, invoice_lines):
+        """ A customer invoice is generated from sale order.
```

Comment:
IMO this first line is misrepresenting the method unless I am misunderstanding. Maybe something like
```
It assigns serial/lot numbers to invoice lines
```